apps/omg_watcher/lib/omg_watcher/db/eth_event.ex

```diff
@@ -72,44 +77,50 @@ defmodule OMG.Watcher.DB.EthEvent do
   @spec insert_exits!([non_neg_integer()]) :: :ok
   def insert_exits!(exits) do
     exits
-    |> Stream.map(&utxo_pos_from_exit_event/1)
+    |> Stream.map(&utxo_exit_from_exit_event/1)
     |> Enum.each(&insert_exit!/1)
   end
 
-  @spec utxo_pos_from_exit_event(%{call_data: %{utxo_pos: pos_integer()}}) :: Utxo.Position.t()
-  defp utxo_pos_from_exit_event(%{call_data: %{utxo_pos: utxo_pos}}), do: Utxo.Position.decode!(utxo_pos)
+  @spec utxo_exit_from_exit_event(%{call_data: %{utxo_pos: pos_integer()}, rootchain_txhash: charlist()}) :: Utxo.Exit.t()
+  defp utxo_exit_from_exit_event(%{call_data: %{utxo_pos: utxo_pos}, rootchain_txhash: rootchain_txhash}) do
+    %{rootchain_txhash: rootchain_txhash, decoded_utxo_position: Utxo.Position.decode!(utxo_pos)}
+  end
+
+  @spec insert_exit!(Utxo.Exit.t()) :: {:ok, %__MODULE__{}} | {:error, Ecto.Changeset.t()}
+  defp insert_exit!(%{rootchain_txhash: rootchain_txhash, decoded_utxo_position: decoded_utxo_position}) do\
+    case get(rootchain_txhash) do
+      nil ->
+        ethevent = %__MODULE__{
+          rootchain_txhash: rootchain_txhash,
+          event_type: :standard_exit,
+        }
 
-  @spec insert_exit!(Utxo.Position.t()) :: {:ok, %__MODULE__{}} | {:error, Ecto.Changeset.t()}
-  defp insert_exit!(Utxo.position(blknum, txindex, _oindex) = position) do
-    {:ok, _} =
-      if existing_exit = get(exit_key(position)) != nil do
+        DB.TxOutput.get_by_position(decoded_utxo_position)
+        |> txoutput_changeset(%{childchain_utxohash: generate_childchain_utxohash(decoded_utxo_position)}, ethevent)
+        |> DB.Repo.update!()
```

Comment:
We're returning "result" tuple here, as below `{:ok, existing_exit}`
```suggestion
        |> DB.Repo.update()
```

Reply:
Ah right. Good catch!

Reply:
Ah, this comment actually makes me think about something related.

The code here updates txoutputs by creating a new ethevent and inserting a row into ethevents and ethevents_txoutputs.

But if I return ethevent here in the code will the txoutputs assocation be loaded into it?

I'm not sure what ecto will do. I'll look into it.

Reply:
If you return the schema Repo.update returned it will have the association.
If you want it when query check `preload` we're using in other queries